propagator/opentelemetry-propagator-ot-trace/src/opentelemetry/propagator/ot_trace/__init__.py

```diff
@@ -0,0 +1,173 @@
+    set_span_in_context,
+)
+from opentelemetry.trace.propagation.textmap import (
+    Getter,
+    Setter,
+    TextMapPropagator,
+    TextMapPropagatorT,
+)
+
+OT_TRACE_ID_HEADER = "ot-tracer-traceid"
+OT_SPAN_ID_HEADER = "ot-tracer-spanid"
+OT_SAMPLED_HEADER = "ot-tracer-sampled"
+OT_BAGGAGE_PREFIX = "ot-baggage-"
+
+_valid_header_name = re_compile(r"^[\w_^`!#$%&'*+.|~]+$")
+_valid_header_value = re_compile(r"^[\t\x20-\x7e\x80-\xff]+$")
+_valid_traceid = re_compile(r"[0-9a-f]{32}|[0-9a-f]{16}")
+_valid_spanid = re_compile(r"[0-9a-f]{16}")
+
+
+class OTTracePropagator(TextMapPropagator):
+    """Propagator for the OTTrace HTTP header format"""
+
+    def extract(
+        self,
+        getter: Getter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: Optional[Context] = None,
+    ) -> Context:
+
+        traceid = extract_first_element(
+            getter.get(carrier, OT_TRACE_ID_HEADER)
+        )
+
+        spanid = extract_first_element(getter.get(carrier, OT_SPAN_ID_HEADER))
+
+        sampled = extract_first_element(getter.get(carrier, OT_SAMPLED_HEADER))
+
+        if sampled == "true":
+            traceflags = TraceFlags.SAMPLED
+        else:
+            traceflags = TraceFlags.DEFAULT
+
+        if (
+            traceid != INVALID_TRACE_ID
+            and _valid_traceid.match(traceid) is not None
+            and spanid != INVALID_SPAN_ID
+            and _valid_spanid.match(spanid) is not None
+        ):
+            context = set_span_in_context(
+                DefaultSpan(
+                    SpanContext(
+                        trace_id=int(traceid, 16),
+                        span_id=int(spanid, 16),
+                        is_remote=True,
+                        trace_flags=traceflags,
+                    )
+                ),
+                context,
+            )
+
+            baggage = get_all(context) or {}
+
+            for key in getter.keys(carrier):
+
+                if not key.startswith(OT_BAGGAGE_PREFIX):
+                    continue
+
+                baggage[key[len(OT_BAGGAGE_PREFIX) :]] = extract_first_element(
+                    getter.get(carrier, key)
+                )
+
+            for key, value in baggage.items():
+                context = set_baggage(key, value, context)
+
+        return context
+
+    def inject(
+        self,
+        set_in_carrier: Setter[TextMapPropagatorT],
+        carrier: TextMapPropagatorT,
+        context: Optional[Context] = None,
+    ) -> None:
+
+        span_context = get_current_span(context).get_span_context()
+
+        if (
+            span_context.trace_id == INVALID_TRACE_ID
+            or _valid_traceid.match(hex(span_context.trace_id)[2:]) is None
+            or span_context.span_id == INVALID_SPAN_ID
+            or _valid_spanid.match(hex(span_context.span_id)[2:]) is None
+        ):
+            return
+
+        set_in_carrier(
+            carrier, OT_TRACE_ID_HEADER, hex(span_context.trace_id)[18:]
```

Comment:
This doesn't seem to work right with trace IDs that don't happen to use all 32 hex digits:

```py
>>> '0xdeadbeef'[18:]
''
>>> '0xffffffffffffffffffffffabc'[18:]
'ffffffabc'
```

Best thing might be use str formatters [like the tracecontext propagator](https://github.com/open-telemetry/opentelemetry-python/blob/24edd3d81afb394c3b2d1f0416faca59db931b66/opentelemetry-api/src/opentelemetry/trace/propagation/tracecontext.py#L98-L99) does and then truncate to 16 hex chars

Reply:
Fixed